streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -1129,9 +1184,31 @@ public void shouldCommitConsumerPositionIfRecordQueueIsEmpty() {
         task.addRecords(partition1, singletonList(getConsumerRecordWithOffsetAsTimestamp(partition1, 0L)));
         task.addRecords(partition2, singletonList(getConsumerRecordWithOffsetAsTimestamp(partition2, 0L)));
         task.process(0L);
+        processorStreamTime.mockProcessor.addProcessorMetadata("key1", 100L);
```

Comment:
The original purpose of this test seems to be testing "partial commits" (ie, for a sub-set of partitions)? Adding metadata make the partial commit a "full commit" and thus seems to change the test?

Als also not sure what the difference of this test compare to the newly added test below is?